pkg/build/strategies/sti/sti.go

```diff
@@ -317,7 +317,6 @@ func (builder *STI) createBuildEnvironment() []string {
 	env, err := scripts.GetEnvironment(builder.config)
 	if err != nil {
 		glog.V(1).Infof("No user environment provided (%v)", err)
-		return nil
```

Comment:
What happens when a legitimate error occurs in GetEnvironment and "env" is returned as nil?  the append is going to fail, right?

you need to at least initialize env if it came back nil.